designcompose/src/main/java/com/android/designcompose/FrameRender.kt

```diff
@@ -150,16 +150,28 @@ private fun calculateRotationData(
 private fun calculateProgressBarData(
     progressBarData: ProgressBarMeterData,
     meterValue: Float,
-    height: Float,
+    style: ViewStyle,
     density: Float
-): Size {
+): Pair<Size, androidx.compose.ui.graphics.Matrix?> {
     // Progress bar discrete values are done by percentage
     val discretizedMeterValue =
         meterValue.coerceDiscrete(progressBarData.discrete, progressBarData.discreteValue)
 
-    // Resize the progress bar by interpolating between 0 and endX
-    val barWidth = lerp(0F, progressBarData.endX, discretizedMeterValue, density)
-    return Size(barWidth, height)
+    // Resize the progress bar by interpolating between 0 and endX or endY depending on whether it
+    // is a horizontal or vertical progress bar
+    if (progressBarData.vertical) {
+        val width = style.width.pointsAsDp(density).value
```

Comment:
This is the same logic as it was before, but I think we should be using layout produced dimensions here, and not consulting style. This approach won't work if the meter is sized using constraints or flexbox, and we're at a point where we have already computed layout, and have the layout info available.

Either we should update in this PR or make a new one to migrate to using layout produced values (I think almost everything in FrameRender should use layout produced values, except the appearance stuff like fill/stroke style -- this is where it would be super nice in the future to tease out ViewStyle into layout bits and presentation bits, because they get consumed at different points).

Reply:
I think the reason I didn't end up using the layout width/height was that those values are calculated after transformation. So a horizontal progress bar with a small height that has been rotated will have a much taller height after the transformation, and layout will return that taller height. Here I need the height before transformation because we apply the transformation to the draw context after determining the size of the rect. Does that make sense? Can you see another way of doing this that isn't too complicated?